omc3/scripts/bad_bpms_summary.py

```diff
@@ -0,0 +1,294 @@
+        "This should be either subfolders in `root` or glob-patterns for those."
+        )
+    )
+    params.add_parameter(
+        name="root",
+        type=PathOrStr,
+        default=ROOT,
+        help="Path to the root directory, containing the dates."
+    )
+    params.add_parameter(
+        name="outfile",
+        type=PathOrStr,
+        help="Path to the file to write out." 
+    )
+    params.add_parameter(
+        name="print_percentage",
+        type=float,
+        help="Print out BPMs that appear in more than this percentage of measurements." 
+    )
+    params.add_parameter(
+        name="accel_glob",
+        type=str,
+        default="LHCB*",
+        help="Accelerator name (glob for the sub-directories)."
+    )
+    return params
+
+
+@entrypoint(get_params(), strict=True)
+def bad_bpms_summary(opt: DotDict):
+    outfile = None
+    if opt.outfile is not None:
+        outfile = Path(opt.outfile)
+        outfile.parent.mkdir(parents=True, exist_ok=True)
+    
+    df_collection = collect_bad_bpms(Path(opt.root), opt.dates, opt.accel_glob)
+    if outfile is not None:
+        tfs.write(outfile.with_stem(f"{outfile.stem}_collected"), df_collection)
+    
+    df_evaluated = evaluate(df_collection)
+    if outfile is not None:
+        tfs.write(outfile, df_evaluated)
+
+    if opt.print_percentage is not None:
+        print_results(df_evaluated, opt.print_percentage)
+
+
+# Collection of Data ---
+
+def get_empty_df() -> tfs.TfsDataFrame:
+    """ Create an empty TfsDataFrame with the correct column names. """
+    return tfs.TfsDataFrame(columns=[NAME, ACCEL, PLANE, SOURCE, FILE])
+
+
+def collect_bad_bpms(root: Path, dates: Sequence[Path | str], accel_glob: str) -> tfs.TfsDataFrame:
+    """ Create a TfsDataFrame with all bad-bpms within selcted dates.
+
+    Args:
+        root (Path): Root path to the GUI output folder.
+        dates (Sequence[Path | str]): List of dates or glob patterns to collect bad-bpms from.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms within selcted dates.
+    
+    """
+    dfs = []
+    for date in dates:
+        date_dir = root / date
+        if date_dir.is_dir():
+            dfs.append(collect_date(date_dir, accel_glob))
+            continue
+
+        for date_dir in root.glob(date):
+            dfs.append(collect_date(date_dir, accel_glob))
+    return tfs.concat(dfs, axis="index", ignore_index=True) 
+
+
+def collect_date(date_dir: Path, accel_glob: str) -> tfs.TfsDataFrame:
+    """ Collect bad-bpms for a single date.
+    
+    Args:
+        date_dir (Path): Path to the date directory.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms for the date.
+    """
+    dfs: list[tfs.TfsDataFrame] = []
+
+    for accel_dir in date_dir.glob(accel_glob):
```

Comment:
This loop seems specific to LHC? Do other machines have the same path like `root/date/accel/measurement`?

Reply:
yes, this structure is specific to the GUI which we use for all accelerators